### ISSUE TITLE
Improve Scripts

### ISSUE THREAD

**Issue by VincenzoFerme** [benchflow/analysers#18]
**TODOs:**
- ~~Have script organised in folder according to the language used to implement them~~
- Correctly organise the configuration files
- [POSTPONED] _Use Dataframe for now, then search for a more performant solution: https://databricks.com/blog/2015/02/17/introducing-dataframes-in-spark-for-large-scale-data-science.html and test user defined functions http://www.datastax.com/dev/blog/cassandra-user-defined-aggregates-python-driver [in Panda style: https://lab.getbase.com/pandarize-spark-dataframes/]. Or: http://ddf.io. first thing to do test performance for DF and RDD for our operations [http://www.agildata.com/apache-spark-rdd-vs-dataframe-vs-dataset/][https://github.com/datastax/spark-cassandra-connector/blob/master/doc/15_python.md]_
- See also http://chapeau.freevariable.com/2014/09/improving-spark-application-performance.html

**PERFORMANCE IMPROVEMENTES:**
- Refactor the scripts to avoid any not necessary duplicate computation, starting from the data on which we work and improving the way we handle it. As for example, when we work with construct, we can pass through them only once and compute all the combination we need.
- Improve the performance of non-Spark (then python) code relying on statistical libraries.
- Any other relevant improvement pointed out by [performance experiments](https://github.com/benchflow/data-transformers/issues/74)

This also applies to https://github.com/benchflow/data-transformers

**Comment by VincenzoFerme** [comment]
@Cerfoglg I want to point out some interesting characteristics and functionality offered by Cassandra that might be useful for simple analysers:
- Aggregates: https://docs.datastax.com/en/cql/3.3/cql/cql_using/useQueryStdAggregate.html
- Custom Aggregates definition: http://christopher-batey.blogspot.ch/2015/05/cassandra-aggregates-min-max-avg-group.html
- At some point, for some analysers, it might be useful to use asynch triggers: https://maxgrinev.wordpress.com/2010/07/23/extending-cassandra-with-asynchronous-triggers/
